docs/19_feature_design_template.md

```diff
@@ -83,20 +83,89 @@ The above questions are focused on one purpose: the place and role of the new fe
 
 # Design
 
+The design section purposes discovering the implementation details for the feature. What is the structure of the feature? How is the feature to be integrated into the project/component(s)? And many other details are covered by the subsections below. But before we dive into explanations for the design parts, one should consider the following rules that hold for the whole design section:
+
+- The design acts within the requirements examined in the [Requirements](#requirements) section.
+- The design details all the [Analysis]($analysis) section describes/touches.
+- The design's subsections must not leave open questions about the selected implementation approach. Each section ends with clear and definitive details about exactly one implementation approach.
+
 ### Architecture
 > Fundamental structures of the feature and context (diagram).
 
+This section examines the overall, top-level architecture of the feature. This architecture defines the approach to the feature implementation discussed in more detail in the next section. The following points hold for the section:
+
+- The section provides and examines the fundamental architecture of the feature with no details (classes, methods, etc.).
+- The section explains the implementation approach to the feature. If there are several implementation approaches:
+    - the section examines each approach and reviews its pros and cons;
+    - the section selects one approach and explains the choice providing clear reasoning.
+- The section represents the architecture with the components diagram demonstrating responsibilities and the overall algorithm.
+
 ### User Interface
 > How users will interact with the feature (API, CLI, Graphical interface, etc.).
 
+In this section, one should explain how users interact with the feature. The important note here is that a user is an abstraction that represents an interaction participant. These could be an end-user, developer, another component, etc. The general purpose is to describe the usage of the feature.
+
+Consider the following examples:
+
+- If the feature is related to the CLI tool, the usage can be described as the feature command executing example or the `help` command output that includes the feature-related points.
+- If the feature is related to the web application, the usage can be described as the User Interface [wireframes](https://plantuml.com/en/salt).
+- If the feature is related to the core implementation, the usage can be represented as the feature integration steps or the feature component usage within another component.
+
 ### Database
```

Comment:
```suggestion
### Data Storage
```
What do you think? I think we discussed this briefly with @solid-vovabeloded as well